notes/plutus-report/applications.tex

```diff
@@ -273,3 +276,188 @@ \subsection{The \glsentrytext{plutus-playground}}
+    \node [place-status]          (committed) [right of=commit,tokens=0,label=above:$\mathsf{committed}$] {};
+    \node [transition]     (reject)  [below of=mempool] {};
+    \node [place-status]          (invalid)  [left of=reject,label=$\mathsf{invalid}$] {};
+
+    \path (mempool) edge [->] (confirm);
+    \path (confirm) edge [->] (confirmed);
+    \path (confirmed) edge [->] (rollback);
+    \path (rollback) edge [->] (mempool);
+    \path (confirmed) edge [->] (commit);
+    \path (commit) edge [->] (committed);
+    \path (mempool) edge [->] (reject);
+    \path (reject) edge [->] (invalid);
+
+  \end{tikzpicture}
+\caption{Petri net for the status of transactions}
+\label{fig:petri-net-txn}
+\end{figure}
+
+\subsubsection{Address change}
+
+The set of unspent outputs at an address is modified by transactions that spend and produce outputs.
+Therefore, whenever the status of a transaction changes, the status of its inputs and outputs changes also.
+
+We represent the outputs at each address with two petri nets, one for unspent outputs and one for spent outputs.
+There is one place each for outputs that are in the mempool, tentatively confirmed, permantently confirmed, rejected.
+
+% TODO: Think about how to do it properly. What level of detail do we need here? Maybe we don't need an extra petri net (address change is just a function of tx change). Or maybe we should include tx dependencies via their outputs as well.
+
+\subsubsection{Endpoint}
+
+Users and other applications may call endpoints on the \gls{app}.
+Endpoints are places $e_1, \ldots, e_n$ in the petri net (see \ref{fig:petri-net-endpoint}).
+
+\begin{figure}
+  \centering
+  \begin{tikzpicture}[node distance=1.3cm,>=stealth',bend angle=45,auto]
+    % Styles are from https://texample.net/tikz/examples/nodetutorial/
+    \tikzstyle{place-endpoint}=[circle,thick,draw=yellow!75,fill=black!20,minimum size=6mm]
+    \tikzstyle{transition}=[rectangle,thick,draw=black!75,
+    fill=black!20,minimum size=4mm]
+
+    \node [place-endpoint,tokens=1] (ep1) [label=left:$e_1$] {};
+    \node [place-endpoint]          (ep2) [right of=ep1,label=below right:$e_2$] {};
+
+  \end{tikzpicture}
+\caption{Petri net for endpoints. The token in $e_1$ signifies that input is available to be consumed by this contract.}
+\label{fig:petri-net-endpoint}
+\end{figure}
+
+\subsubsection{Apps}
+
+Given the places for time, transaction status and endpoints we can describe \glspl{app} as sequences of transitions.
+The states of the app are represened by places.
+\ref{fig:plutus-app-net} shows a \gls{app} with three possible state, $s_1$, $s_2$ and $s_3$.
+As soon as a transaction is confirmed, the state can progress from $s_1$ to $s_2$.
+
+After that, endpoint $e_1$ becomes \emph{active}, meaning that the app can make progress as soon as input is provided. The next state depends on which of two possible events happens first: The endpoint being called by the user, or the clock reaching slot ten.
```

Comment:
Is this right? In the petri net that you drew, I think it can still move a token into s3 even if the move to s4 has already happened. That is, I'm not sure how you do a *race* where only one of two things can happen, depending on which condition you get to first. Maybe you have a shared place that they draw from with one token that they both need?

Reply:
Yeah when a token is in s2 then only one of the two transitions can fire, because it takes out the token.

Reply:
Oh right of course.